### PR TITLE
 Add a 'Table of Content' to the settingUp.md #5112 

### LINKED ISSUE #5112: Add a 'Table of Content' to the settingUp.md
Add a hyperlinked table of content at the top of `/devdocs/settingUp.md`

### PR DESCRIPTION
**Pull request by dustinlapierre**
Fixes #5112

### CODE REVIEW

**Comment by damithc** [comment]
The PR description (i.e. first comment) should contain `Fixes Issue_Number` so that the corresponding issue gets auto-closed when we merge this PR. You can refer other PRs for an example.

File: devdocs/settingUp.md [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+##<a name="1"></a>Setting Up the Developer Environment
 >If you encounter any problems during the setting up process, please refer to our [troubleshooting guide](troubleshooting-guide.md) before posting a help request in our [issue tracker](https://github.com/TEAMMATES/teammates/issues).
 
 These instructions work for Linux, OS X as well as for the Windows
 environment. The only difference for Windows environment is that the command `./gradlew` should be replaced by `gradlew.bat` everywhere.
 
 The full tool stack is given at the [end of this document](#toolStack).
 
-##Prerequisites
+##<a name="2"></a>Prerequisites
```
Comment (damithc) [inline devdocs/settingUp.md]: There is no need to add these. Refer this for an example https://github.com/TEAMMATES/teammates/edit/master/devdocs/orientation-guide.md

File: devdocs/settingUp.md [hunk 2]
```diff
@@ -1,12 +1,24 @@
-#Setting Up the Developer Environment
+#Table of Contents
+1. [Setting Up the Developer Environment](#1)
+2. [Prerequisites](#2)<br>
+3. [Setting up the dev server](#3)
+4. [Running the test suite](#4)
+5. [Deploying to a staging server](#5)
+6. [Running client scripts](#6)
+7. [Troubleshooting](#7)
+8. [Tool stack](#toolStack)
```
Comment (damithc) [inline devdocs/settingUp.md]: No need for numbers.

**Comment by damithc** [comment]
Added some comments for you to work on @dustinlapierre

**Comment by dustinlapierre** [comment]
ok I have all the changes you asked for made I'll make the new PR now

**Comment by dustinlapierre** [comment]
hopefully everything is up to code now

File: devdocs/settingUp.md [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+* [Prerequisites](#prerequisites)
+* [Setting up the dev server](#setting-up-the-dev-server)
+* [Running the test suite](#running-the-test-suite)
+* [Deploying to a staging server](#deploying-to-a-staging-server)
+* [Running client scripts](#running-client-scripts)
+* [Troubleshooting](#troubleshooting)
+* [Tool stack](#toolStack)
+
+
+
```
Comment (damithc) [inline devdocs/settingUp.md]: Too many blank lines here?

**Comment by damithc** [comment]
One more minor change requested.

**Comment by dustinlapierre** [comment]
Not sure if your last comment was referring to the white space, but that change is made

**Comment by damithc** [comment]
Thanks for your contribution @dustinlapierre :+1: 
Things to improve if you submit more PRs (note that you should not do more than one `d.firstTimers` issues), 
1. when naming the branch, pick keywords from the issue name. e.g. `add` instead of `added`
2. squash unnecessary commits. The branch has too many commits.
3. Do not change things not related to the PR. e.g. adding a space after `##`

**Pull request merged**